c_src/workitems.cc

```diff
@@ -791,9 +812,9 @@ work_result RangeScanTask::operator()()
                     //------------------------------------------------------------
 
                     try {
-                        options_.range_filter_ = 
+                        range_filter_ = 
                             parse_range_filter_opts(options_.env_, 
-                                                    options_.rangeFilterSpec_, *(options_.extractor_),
+                                                    options_.rangeFilterSpec_, *(extractor_),
                                                     throwIfBadFilter);
                     } catch(std::runtime_error& err) {
```

Comment:
This is going to throw.  I believe it is throwing to threading.cc notify_caller() which is called by eleveldb_write_thread_worker().  I do not see try/catch block anywhere in that call stack.